halpybot/packages/board/board.py

```diff
@@ -120,9 +174,9 @@ def _update_last_index(self):
 
     def return_rescue(self, key: typing.Union[str, int]) -> Case:
         """Find a Case given the Client Name or Case ID"""
-        if isinstance(key, str):
-            return self._cases_by_id[self._case_alias_name[key.casefold()]]
-        if isinstance(key, int):
+        if isinstance(key, str) and key in self._case_alias_name:
+            return self._cases_by_id[self._case_alias_name[key]]
+        if isinstance(key, int) and key in self._cases_by_id:
```

Comment:
these added checks seem unnecessary, given the guarded code is going to raise a KeyError if they don't exist.